### PR TITLE
WasmFS: Add nanosecond support to WasmFS

### PR DESCRIPTION
**Pull request by jameshu15869**
This PR adds nanosecond support for file times in WasmFS, specifically when used for `utime` and `stat`. This PR was separated from https://github.com/emscripten-core/emscripten/pull/19561#discussion_r1229927897, reusing the same tests.

### CODE REVIEW

**Comment by tlively** [comment]
It looks like the diff still contains changes from #19561. If you change the target branch for this PR from `main` to `jameshu15869:library-wasmfs-utime`, that would fix the diff. Although it looks like that branch is in your fork, so this probably isn't possible, unfortunately. We'll have to wait until #19561 is merged to get a reasonable diff here.

**Comment by jameshu15869** [comment]
Ah I see. For this PR, I did `git checkout -b` on my old (combined) branch before manually undoing the NS changes on https://github.com/emscripten-core/emscripten/pull/19561. Is there a better way to do this in the future?

**Comment by tlively** [comment]
One way would be to add a commit undoing the changes on the original PR, then fork a new branch off of it, then revert the undoing of the changes on the new branch so that the revert commit actually reapplies the changes. Then even if the diff for the PR that adds the changes includes everything, we can just look at the diff for that revert commit to see only the new changes.

File: system/lib/wasmfs/syscalls.cpp
Comment (kripken) [inline system/lib/wasmfs/syscalls.cpp:1097]: this TODO can be removed now

**Review by kripken — APPROVED** [review]
Code lgtm but we have some failures on `main` it seems, that are showing up here and everywhere else. Work is ongoing to fix those but this will need to wait to land on that.

**Comment by sbc100** [comment]
Rebasing or merging should fix the CI issues.

**Pull request merged**
